src/toxcore/messenger/conference/packet/conference_action.rs

```diff
@@ -0,0 +1,93 @@
+/*! Conference action action struct.
+*/
+
+use std::str;
+use nom::{be_u16, be_u32, rest};
+
+use crate::toxcore::binary_io::*;
+
+/** ConferenceAction is the struct that holds info to send action to a conference.
+
+Sent to notify action to all member of conference.
+
+Serialized form:
+
+Length    | Content
+--------- | ------
+`1`       | `0x63`
+`2`       | `conference number`
+`2`       | `peer number`
+`4`       | `action number`
+`1`       | `0x41`
+variable  | `action`(UTF-8 C String)
+
+*/
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct ConferenceAction {
```

Comment:
No need to prefix the name with `Conference`, same for the file name

Reply:
```
error[E0659]: `Action` is ambiguous (glob import vs glob import in the same module)
  --> src/toxcore/messenger/packet/mod.rs:41:12
   |
41 |     Action(Action),
   |            ^^^^^^ ambiguous name
   |
note: `Action` could refer to the struct imported here
  --> src/toxcore/messenger/packet/mod.rs:20:9
   |
20 | pub use self::action::*;
   |         ^^^^^^^^^^^^^^^
   = help: consider adding an explicit import of `Action` to disambiguate
note: `Action` could also refer to the struct imported here
  --> src/toxcore/messenger/packet/mod.rs:32:85
   |
32 | pub use crate::toxcore::messenger::conference::packet::{Packet as ConferencePacket, *};
   |                                                                                     ^
   = help: consider adding an explicit import of `Action` to disambiguate
```

Reply:
that's because of `pub use crate::toxcore::messenger::conference::packet::{Packet as ConferencePacket, *};` . You don't need to include them all , only `crate::toxcore::messenger::conference::packet::{Packet as ConferencePacket`